frontend-react/src/components/ManagePublicKey/ManagePublicKey.test.tsx

```diff
@@ -0,0 +1,87 @@
+import { screen } from "@testing-library/react";
+
+import { UseOrganizationSendersResult } from "../../hooks/UseOrganizationSenders";
+import { RSSender } from "../../config/endpoints/settings";
+import { renderApp } from "../../utils/CustomRenderUtils";
+import * as useOrganizationSendersExports from "../../hooks/UseOrganizationSenders";
+
+import { ManagePublicKey } from "./ManagePublicKey";
+
+const DEFAULT_SENDERS: RSSender[] = [
+    {
+        name: "default",
+        organizationName: "ignore",
+        format: "CSV",
+        customerStatus: "inactive",
+        schemaName: "primedatainput/pdi-covid-19",
+        processingType: "sync",
+        allowDuplicates: true,
+        topic: "covid-19",
+    },
+    {
+        name: "ignore-full-elr",
+        organizationName: "ignore",
+        format: "HL7",
+        customerStatus: "active",
+        schemaName: "strac/strac-covid-19",
+        processingType: "sync",
+        allowDuplicates: true,
+        topic: "full-elr",
+    },
+];
+const contentString = "foo,bar\r\nbar,foo";
+
+const fakeFile = new File([new Blob([contentString])], "file.pem", {
+    type: "application/x-x509-ca-cert",
+});
+fakeFile.text = () => Promise.resolve(contentString);
+
+describe("ManagePublicKey", () => {
+    afterEach(() => {
+        jest.restoreAllMocks();
+    });
+
+    function mockUseOrganizationSenders(
+        result: Partial<UseOrganizationSendersResult> = {}
+    ) {
+        jest.spyOn(
+            useOrganizationSendersExports,
+            "UseOrganizationSenders"
+        ).mockReturnValue({
+            isLoading: false,
+            senders: DEFAULT_SENDERS,
+            ...result,
+        });
+    }
+
+    describe("when the sender options are loading", () => {
```

Comment:
It's also not entirely important to check each individual step's contents.  Since ManagePublicKey is the "manager" component of all its children, you could even do something like:

```tsx
jest.mock("./ManagePublicKeyUpload", () => {
    ManagePublicKeyUpload: () => <div data-testid="ManagePublicKeyUpload" />
});

// then just verify that the test id exists in the dom
expect(screen.getByTestId("ManagePublicKeyUpload")).toBeVisible();
```

Each individual step component should have its own accompanying tests so it's irrelevant for the manager component to test that as well specifically for the state --> step checks.  However, mocking individual components like this would prevent testing the entire flow through an integration test so that's something to consider.

Reply:
Definitely was intended to add more test cases. Was waiting on feedback before proceeding.